client/state/data-getters/index.js

```diff
@@ -91,6 +91,55 @@ export const requestActivityLogs = ( siteId, filter, { freshness = 5 * 60 * 1000
 	);
 };
 
+const requestExternalContributorsId = siteId => `site-external-contributors-${ siteId }`;
+
+export const requestExternalContributors = siteId =>
+	requestHttpData(
+		requestExternalContributorsId( siteId ),
+		http( {
+			method: 'GET',
+			path: `/sites/${ siteId }/external-contributors`,
+			apiNamespace: 'wpcom/v2',
+		} ),
+		{
+			fromApi: () => data => [ [ requestExternalContributorsId( siteId ), data ] ],
+		}
+	);
+
+export const requestExternalContributorsAddition = ( siteId, userId ) => {
+	const requestId = requestExternalContributorsId( siteId );
+	const id = `${ requestId }-addition-${ userId }`;
+	return requestHttpData(
+		id,
+		http( {
+			method: 'POST',
+			path: `/sites/${ siteId }/external-contributors/add`,
+			apiNamespace: 'wpcom/v2',
+			body: { user_id: userId },
+		} ),
+		{
+			fromApi: () => data => [ [ requestId, data ] ],
+		}
+	);
+};
+
+export const requestExternalContributorsRemoval = ( siteId, userId ) => {
+	const requestId = requestExternalContributorsId( siteId );
+	const id = `${ requestId }-removal-${ userId }`;
+	return requestHttpData(
+		id,
+		http( {
+			method: 'POST',
+			path: `/sites/${ siteId }/external-contributors/remove`,
+			apiNamespace: 'wpcom/v2',
+			body: { user_id: userId },
+		} ),
+		{
+			fromApi: () => data => [ [ requestId, data ] ],
+		}
+	);
+};
+
```

Comment:
there's a subtle bug in these `POST` functions I didn't mention before and I don't want to block this PR, but I want you to be aware that it's there

the default `freshness` is `Infinity` which means that after the first unique request goes out no more will send even if we reissue them. that is, we'll only be able to remove a contributor _once_ - if someone removes, adds, then removes the same contractor (by id) then it will fail to remove them the second time.

thankfully if this happens they only need to reload their browsers because `httpData` doesn't yet persist its data

when it comes time to fix this bug we'll add freshness and that can be in a data-dependent way if we need, but it gets more complicated. I won't write the code out here because again, I don't want to block this PR or throw too much your way. nonetheless we'd need to reset the data or invalidate it before the system will actually re-send the same request